docs/proposals/008-tuples.md

```diff
@@ -0,0 +1,137 @@
+SP #008 - Tuples
+==============
+
+Now that we have variadic generics in the language following [SP #007], we should now be able to support `Tuple` type as a core language feature.
+`Tuple` types are useful in many places to reduce boilerplate in user code, such as in function return types to eliminate the need of defining
+`struct`s that are used only for invoking the function. Adding `Tuple` types to Slang will also simplify interop with other languages such as Python
+and C++ that have tuple types.
+
+Status
+------
+
+Author: Yong He
+
+Implementation: In-progress.
+
+Reviewed by: N/A
+
+Background
+----------
+
+Tuple type is widely supported in almost all of the modern programming languages including C++, C#, Swift, Rust, Python. Supporting tuple types
+in Slang will bring the language to parity with other languages and allow users to practice the same coding idioms in Slang, and allow Slang code
+to interop more directly with other parts of the user application written in other languages.
+
+
+Proposed Approach
+-----------------
+
+With variadic generics support, we can now easily define a Tuple type in stdlib as:
+```
+__generic<each T>
+__magic_type(TupleType)
+struct Tuple
```

Comment:
How would the members be declared?
Should it be something like a following?
```
__generic<T0, each T>
struct Tuple : Tuple<T>
{
     T0 e0;
};
```

Reply:
We can't define the members inside the stdlib, just like we can't define `x`, `y` etc in a vector type, it has to be implemented as a special syntax in the parser.